File: src/org/cappuccino_project/ide/intellij/plugin/indices/IndexKeyUtil.java
package org.cappuccino_project.ide.intellij.plugin.indices;

import com.intellij.psi.PsiElement;
import com.intellij.psi.stubs.StubIndexExtension;
import com.intellij.psi.stubs.StubIndexKey;
import org.jetbrains.annotations.NotNull;

public class IndexKeyUtil {
    private IndexKeyUtil(){}
    @NotNull
    public static <K, Psi extends PsiElement> StubIndexKey<K, Psi> createIndexKey(@NotNull Class<? extends StubIndexExtension<K, Psi>> indexClass) {
        return StubIndexKey.createIndexKey(indexClass.getCanonicalName());
    }
}

File: src/org/cappuccino_project/ide/intellij/plugin/indices/IndexKeyUtil.kt
package org.cappuccino_project.ide.intellij.plugin.indices

import com.intellij.psi.PsiElement
import com.intellij.psi.stubs.StubIndexExtension
import com.intellij.psi.stubs.StubIndexKey

object IndexKeyUtil {
    fun <K, Psi : PsiElement> createIndexKey(indexClass: Class<out StubIndexExtension<K, Psi>>): StubIndexKey<K, Psi> {
        return StubIndexKey.createIndexKey(indexClass.canonicalName)
    }
}
